### PR TITLE
[Improve][core/transformv2]flink transform v2 use richflatmap instead flatmap

### PR DESCRIPTION
**Pull request by tobezhou33**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [x] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [x] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Please merge dev into your branch, thanks!

**Comment by tobezhou33** [comment]
> Please merge dev into your branch, thanks!

done, pls help  me review，thanks! @TyrantLucifer

**Review by Hisoka-X — DISMISSED** [review]
LGTM

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java [hunk 1]
```diff
@@ -120,7 +121,19 @@ protected DataStream<Row> flinkTransform(SeaTunnelTransform transform, DataStrea
                 new FlinkRowConverter(transform.getProducedType());
         DataStream<Row> output =
                 stream.flatMap(
-                        new FlatMapFunction<Row, Row>() {
+                        new RichFlatMapFunction<Row, Row>() {
+                            @Override
+                            public void open(Configuration parameters) throws Exception {
+                                super.open(parameters);
```
Comment (TyrantLucifer) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java]: Why add this? Could you please explain the meaning of this?

Comment (tobezhou33) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java:127]: the SeaTunnelTransform interface default has func open() and close() when transform initialed and completed，but FlatMapFunction dont have open and close func ,when user define transform have do sth in open() or close(), the FlatMapFunction cant satisfy， such as SQLTransform。

Comment (TyrantLucifer) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java]: I mean that `super.open`

Comment (tobezhou33) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-starter-common/src/main/java/org/apache/seatunnel/core/starter/flink/execution/TransformExecuteProcessor.java]: Sorry, I misunderstood the meaning，this super.open can be removed。it default empty

**Pull request closed without merging**